From d69346bf7d3b2dc32fdfc2fbdca7be70570db0d1 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Thu, 22 Dec 2016 12:17:32 +0100
Subject: [PATCH 0674/1345] telephony: mvebu: use common macro for stop
 polling timeout

commit  a98769db9031259ed6e04abc504c4bf68df06474 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

This commit also removes magic from tdm2c_pcm_disable.

Change-Id: I10c356f9c15e8627f0d3fb914b6190f0b23d7122
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/35043
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/telephony/mvebu_phone/mv_phone_dev.c |   13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/drivers/telephony/mvebu_phone/mv_phone_dev.c b/drivers/telephony/mvebu_phone/mv_phone_dev.c
index 83899e9..07e11c6 100644
--- a/drivers/telephony/mvebu_phone/mv_phone_dev.c
+++ b/drivers/telephony/mvebu_phone/mv_phone_dev.c
@@ -112,8 +112,6 @@
 int mv_phone_enabled;
 struct mv_phone_dev *priv;
 
-#define TDM_STOP_MAX_POLLING_TIME 20 /* ms */
-
 /* TDM Interrupt Service Routine */
 static irqreturn_t tdm_if_isr(int irq, void *dev_id);
 
@@ -213,7 +211,7 @@ static void tdm2c_if_pcm_start(void)
 		tdm2c_pcm_start();
 	} else {
 		pcm_start_stop_state++;
-		while (is_pcm_stopping && max_poll < TDM_STOP_MAX_POLLING_TIME) {
+		while (is_pcm_stopping && max_poll < MV_TDM_STOP_POLLING_TIMEOUT) {
 			spin_unlock_irqrestore(&tdm_if_lock, flags);
 			mdelay(1);
 			max_poll++;
@@ -427,20 +425,21 @@ int tdm_if_init(struct tal_params *tal_params)
 	return ret;
 }
 
+/* Disable TDM2C PCM */
 void tdm2c_pcm_disable(void)
 {
 	u32 max_poll = 0;
 
 	tdm2c_if_pcm_stop();
 
-	while ((is_pcm_stopping != 0) && (max_poll < 20)) {
+	while ((is_pcm_stopping != 0) && (max_poll < MV_TDM_STOP_POLLING_TIMEOUT)) {
 		mdelay(1);
 		max_poll++;
 	}
 
-	if (max_poll >= 20)
-		dev_warn(priv->dev, "%s: stopping pcm channels exceeded 20ms\n",
-			 __func__);
+	if (max_poll >= MV_TDM_STOP_POLLING_TIMEOUT)
+		dev_warn(priv->dev, "\n%s: Channels disabling timeout (%dms)\n",
+			 __func__, MV_TDM_STOP_POLLING_TIMEOUT);
 
 }
 
-- 
1.7.9.5

